src/model/entities/part.ts

```diff
@@ -292,7 +294,11 @@ export class Part {
   public reset(): void {
     this.executedAt = 0
     this.playedDuration = 0
-    this.pieces = this.pieces.filter(piece => piece.isPlanned)
+    this.pieces = [
+      ...this.pieces.filter(piece => piece.isPlanned),
+      ...this.replacedPlannedPieces
+    ]
```

Comment:
Does the order of the pieces matter?

Reply:
For Pieces, no, the order doesn't matter. They don't have any attributes to indicate order.
Pieces have a `start` and `duration` and are "just placed" on the Timeline. Then the TimelineObjects battle it out to see who wins.